delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -144,7 +149,10 @@ def _prepare_customer(self, picking):
 
         """
         company = picking.company_id
-        partner = company.partner_id
+        if picking.picking_type_id.code == "outgoing":
+            partner = company.partner_id
+        elif picking.picking_type_id.code == "incoming":
+            partner = picking.partner_id
```

Comment:
Same remark. Add new method to return the relevant partner to not copy the same logic.